src/PIL/DdsImagePlugin.py

```diff
@@ -121,131 +289,165 @@ def _open(self):
+
+    # fmt: on
 
     def load_seek(self, pos):
         pass
 
 
+# fmt: off
 def _save(im, fp, filename):
-    if im.mode not in ("RGB", "RGBA"):
+    if im.mode not in ("RGB", "RGBA", "L", 'LA'):
         raise OSError(f"cannot write mode {im.mode} as DDS")
 
+    pixel_flags = DDPF.RGB
+    if im.mode == "RGB":
+        rgba_mask = struct.pack("<4I", 0x00FF0000, 0x0000FF00, 0x000000FF, 0x00000000)
+        bit_count = 24
+    elif im.mode == "RGBA":
+        pixel_flags |= DDPF.ALPHAPIXELS
+        rgba_mask = struct.pack("<4I", 0x00FF0000, 0x0000FF00, 0x000000FF, 0xFF000000)
+        bit_count = 32
+        r, g, b, a = im.split()
+        im = Image.merge("RGBA", (a, r, g, b))
+    elif im.mode == 'LA':
+        pixel_flags = DDPF.LUMINANCE | DDPF.ALPHAPIXELS
+        rgba_mask = struct.pack("<4I", 0x000000FF, 0x000000FF, 0x000000FF, 0x0000FF00)
+        bit_count = 16
+    else:  # im.mode == "L"
+        pixel_flags = DDPF.LUMINANCE
+        rgba_mask = struct.pack("<4I", 0xFF000000, 0xFF000000, 0xFF000000, 0x00000000)
+        bit_count = 8
+
+    flags = DDSD.CAPS | DDSD.HEIGHT | DDSD.WIDTH | DDSD.PITCH | DDSD.PIXELFORMAT
+
+    stride = (im.width * bit_count + 7) // 8
     fp.write(
         o32(DDS_MAGIC)
-        + o32(124)  # header size
-        + o32(
-            DDSD_CAPS | DDSD_HEIGHT | DDSD_WIDTH | DDSD_PITCH | DDSD_PIXELFORMAT
-        )  # flags
-        + o32(im.height)
-        + o32(im.width)
-        + o32((im.width * (32 if im.mode == "RGBA" else 24) + 7) // 8)  # pitch
-        + o32(0)  # depth
-        + o32(0)  # mipmaps
-        + o32(0) * 11  # reserved
-        + o32(32)  # pfsize
-        + o32(DDS_RGBA if im.mode == "RGBA" else DDPF_RGB)  # pfflags
-        + o32(0)  # fourcc
-        + o32(32 if im.mode == "RGBA" else 24)  # bitcount
-        + o32(0xFF0000)  # rbitmask
-        + o32(0xFF00)  # gbitmask
-        + o32(0xFF)  # bbitmask
-        + o32(0xFF000000 if im.mode == "RGBA" else 0)  # abitmask
-        + o32(DDSCAPS_TEXTURE)  # dwCaps
-        + o32(0)  # dwCaps2
-        + o32(0)  # dwCaps3
-        + o32(0)  # dwCaps4
-        + o32(0)  # dwReserved2
+        # header size, flags, height, width, pith, depth, mipmaps
+        + struct.pack("<IIIIIII", 124, flags, im.height, im.width, stride, 0, 0, )
+        + struct.pack("11I", *((0,) * 11))  # reserved
+        # pfsize, pfflags, fourcc, bitcount
+        + struct.pack("<IIII", 32, pixel_flags, 0, bit_count)
+        + rgba_mask  # dwRGBABitMask
+        + struct.pack("<IIIII", DDSCAPS.TEXTURE, 0, 0, 0, 0)
     )
-    if im.mode == "RGBA":
-        r, g, b, a = im.split()
-        im = Image.merge("RGBA", (a, r, g, b))
-    ImageFile._save(im, fp, [("raw", (0, 0) + im.size, 0, (im.mode[::-1], 0, 1))])
+    if im.mode == 'LA':
+        ImageFile._save(im, fp, [Image.Tile("raw", (0, 0) + im.size, 0, ('LA', 0, 1))])
+    else:
+        ImageFile._save(im, fp, [Image.Tile("raw", (0, 0) + im.size, 0, (im.mode[::-1], 0, 1))])
```

Comment:
```suggestion
    mode = "LA" if im.mode == "LA" else im.mode[::-1]
    ImageFile._save(im, fp, [Image.Tile("raw", (0, 0) + im.size, 0, (mode, 0, 1))])
```
This is simpler, and should fix the lint error.